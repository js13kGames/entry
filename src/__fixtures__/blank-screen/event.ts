import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 1,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/blank-screen",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "360b838";
