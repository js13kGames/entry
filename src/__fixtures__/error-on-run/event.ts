import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 6,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/error-on-run",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;
