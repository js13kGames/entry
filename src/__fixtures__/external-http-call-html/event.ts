import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 5,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/external-http-call-html",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "c9a43c6";
