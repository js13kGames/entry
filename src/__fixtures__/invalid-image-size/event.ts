import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 12,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/invalid-image-size",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "ce740ea";
