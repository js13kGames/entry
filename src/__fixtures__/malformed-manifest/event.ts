import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 10,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/malformed-manifest",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;
