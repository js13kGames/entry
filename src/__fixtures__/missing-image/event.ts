import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 11,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-image",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "400d963";
