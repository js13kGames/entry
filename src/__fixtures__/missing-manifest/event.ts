import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 9,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-manifest",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "de7f563";
