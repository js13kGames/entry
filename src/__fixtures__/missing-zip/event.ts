import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 3,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-zip",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "e60be98";
