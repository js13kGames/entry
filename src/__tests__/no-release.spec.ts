import * as event from "../__fixtures__/no-release/event";
import { create, GithubClient } from "../services/github";
import { generateReport } from "../report";
import { setComment } from "../comment";
import { getLatestRelease } from "../getLatestRelease";

jest.setTimeout(40000);

describe("integration  no-release", () => {
  const { pullRequest, installation } = event;

  let github: GithubClient;

  it("should init github client", async () => {
    github = await create(installation.id);

    expect(github).toBeTruthy();
  });

  it("should get the latest release", async () => {
    const re = await getLatestRelease({ github })(pullRequest);

    expect(!re).toBeTruthy();
  });

  it("should generate report", () => {
    const report = generateReport();

    expect(report).toMatchSnapshot();
  });

  it("should report", async () => {
    await setComment({ github })(pullRequest, generateReport());
  });
});
