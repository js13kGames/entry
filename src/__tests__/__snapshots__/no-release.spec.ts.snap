// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration  no-release should analyze 1`] = `
Array [
  Object {
    "conclusion": "success",
    "files": Array [
      "image_large.jpg",
      "image_thumbnail.jpg",
      "manifest.json",
    ],
    "name": "manifest-found",
  },
  Object {
    "conclusion": "success",
    "manifest": Object {
      "description": "This entry should yield an error because there is no relese on the pullrequest branch",
      "name": "no release",
    },
    "name": "manifest-read",
  },
  Object {
    "conclusion": "success",
    "gameName": "no release",
    "name": "name-found",
  },
  Object {
    "conclusion": "success",
    "gameDescription": "This entry should yield an error because there is no relese on the pullrequest branch",
    "name": "description-found",
  },
  Object {
    "conclusion": "success",
    "images": Object {
      "image_large": "https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_200,w_300/81d8ad1eef5dbf6feda2665a0df3bf19.jpg",
      "image_thumbnail": "https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_100,w_100/61191dccc4c29b4d785cae9b2f2eae2d.jpg",
    },
    "name": "images-found",
  },
  Object {
    "branch": "entry/no-release",
    "conclusion": "failure",
    "name": "release-found",
    "releaseUrl": undefined,
    "tagName": undefined,
  },
]
`;

exports[`integration  no-release should generate checks 1`] = `
Array [
  Object {
    "conclusion": "success",
    "name": "meta",
    "output": Object {
      "summary": "
## ‚úîÔ∏è manifest found
The pull requet contains a file named \`manifest.json\`

## ‚úîÔ∏è manifest parsed
The manifest was found :
\`\`\`json
{
  \\"name\\": \\"no release\\",
  \\"description\\": \\"This entry should yield an error because there is no relese on the pullrequest branch\\"
}
\`\`\`

## ‚úîÔ∏è game info
The game name is \\"no release\\"

The game description is:

This entry should yield an error because there is no relese on the pullrequest branch

## ‚úîÔ∏è game images
 - image_thumbnail

<img height=\\"100\\" src=\\"https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_100,w_100/61191dccc4c29b4d785cae9b2f2eae2d.jpg\\" />
 - image_large

<img height=\\"100\\" src=\\"https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_200,w_300/81d8ad1eef5dbf6feda2665a0df3bf19.jpg\\" />",
      "text": "",
      "title": "",
    },
  },
  Object {
    "conclusion": "failure",
    "name": "bundle",
    "output": Object {
      "summary": "
## ‚ùå release
The branch of the pull request undefined does not contain any release.

> Submit a release on undefined which contains your bundle.zip as asset.
> Read this [guide](.) if you are stuck",
      "text": "",
      "title": "",
    },
  },
  Object {
    "conclusion": "failure",
    "name": "game",
    "output": Object {
      "summary": "
## ‚ùå deploy
The game could not be deployed, check the bundle report to see why.

> Make sure that the bundle is valid first.",
      "text": "",
      "title": "",
    },
  },
]
`;

exports[`integration  no-release should generate report 1`] = `
"Hello üëã
I am a bot. My job is to review your submission.

Please create a release with your bundled zip file so I can start reviewing üëÄ
If you have trouble doing so, please read the [guide](https://github.com/js13kGames/bot/blob/master/doc/how-to-submit.md#release). Or ask for help here.



<blockquote> <br> <img height=\\"80\\" width=\\"80\\" title=\\"thumbnail\\" src=\\"https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_100,w_100/61191dccc4c29b4d785cae9b2f2eae2d.jpg\\" /> <img height=\\"80\\" title=\\"large image\\" src=\\"https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_200,w_300/81d8ad1eef5dbf6feda2665a0df3bf19.jpg\\" /> <div>  <a href=\\"undefined\\">   <strong>no release</strong>  </a>  <br>  This entry should yield an error because there is no relese on the pullrequest branch </div> <br></blockquote>"
`;
